### PR TITLE
CPM-616 Add setImageValue UserIntent

### PR DESCRIPTION
**Pull request by amigaire**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [x] Tests

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php [hunk 1]
```diff
@@ -1009,6 +1009,80 @@ public function it_creates_and_updates_a_product_with_a_file_value(): void
...
+        $anImagePath = $this->getFileInfoKey($this->getFixturePath('akeneo.png'));
+        $command = new UpsertProductCommand(
+            userId: $this->getUserId('admin'),
+            productIdentifier: 'identifier',
+            valueUserIntents: [new SetFileValue('a_file', null, null, $anImagePath)]
+        );
+        $this->messageBus->dispatch($command);
+
+        $this->clearDoctrineUoW();
+    }
```
Comment (amigaire) [inline src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1031]: Un oublie de la carte CPM-615

**Review by nmarniesse — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php [hunk 2]
```diff
@@ -1009,6 +1009,80 @@ public function it_creates_and_updates_a_product_with_a_file_value(): void
...
+        // create product 'identifier'
+        $anImagePath = $this->getFileInfoKey($this->getFixturePath('akeneo.png'));
+        $command = new UpsertProductCommand(
+            userId: $this->getUserId('admin'),
+            productIdentifier: 'identifier',
+            valueUserIntents: [new SetFileValue('a_file', null, null, $anImagePath)]
+        );
+        $this->messageBus->dispatch($command);
+
+        $this->clearDoctrineUoW();
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1030]: you can use `$this->createProduct`

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php [hunk 3]
```diff
@@ -1009,6 +1009,80 @@ public function it_creates_and_updates_a_product_with_a_file_value(): void
...
+        // create product 'identifier'
+        $anImagePath = $this->getFileInfoKey($this->getFixturePath('akeneo.png'));
+        $command = new UpsertProductCommand(
+            userId: $this->getUserId('admin'),
+            productIdentifier: 'identifier',
+            valueUserIntents: [new SetImageValue('an_image', null, null, $anImagePath)]
+        );
+        $this->messageBus->dispatch($command);
+
+        $this->clearDoctrineUoW();
```
Comment (cemorin) [inline src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1048]: same here

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php (hunk 2 above)
Comment (amigaire) [reply src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1030]: Yes, I changed it

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php (hunk 3 above)
Comment (amigaire) [reply src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php]: Here I want to check that when the command is used to create a product, it throws an error.
So I must use the message bus

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php (hunk 2 above)
Comment (amigaire) [reply src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php:1030]: After discussion, the test check the creation and the update of the product with the command. So we keep the use of the command

**Review by cemorin — APPROVED** [review]

**Pull request merged**